### ISSUE TITLE
Guvectorize signature with frozen arguments

### ISSUE THREAD

**Issue by albop** [numba/numba#1668]
Here is a simple function:
```
def fun(x,y,out):
    out[0] = x[0]*y[0]
    out[1] = x[0]-y[1]
    out[2] = x[1]*y[0]
```
If would be great if this kind of function could be guvectorized by:
```
ftylist = [float64[:](float64[:],float64[:])]
signature = '(p),(q)->(3)'
# signature = '(p),(q)->(r)` # does not work because size of output is unknown
gufun = guvectorize(ftylist=ftylist, signature=signature)(fun)
out = gufun(x,y)
```
In the numpy mailing list there was discussion about whether this kind of signature should be part of the gufun standard and whether it could be extended to more general formulas where the size of outputs is a function of inputs (such as `(n),(n)->(n*(n-1)/2)`) (https://mail.scipy.org/pipermail/numpy-discussion/2015-June/073108.html). Without going that far, support for frozen dimensions would be uncontroversial and useful (and potentially open the way for more optimizations).

**Comment by seibert** [comment]
Agreed.  We are definitely interested in ways to expand the semantics of gufuncs.

**Comment by shoyer** [comment]
This has been implemented upstream in NumPy 1.16: https://www.numpy.org/neps/nep-0020-gufunc-signature-enhancement.html

**Comment by chipkent** [comment]
The latest generalized universal function numpy API does seem to support frozen sizes.  Is there anything blocking this from being implemented in numba?
https://numpy.org/doc/stable/reference/c-api/generalized-ufuncs.html#generalized-universal-function-api

**Comment by daultmcc-sbu** [comment]
Has there been any progress on this?
